winch/codegen/src/codegen/mod.rs

```diff
@@ -1295,6 +1295,31 @@ where
 
         Ok(())
     }
+
+    pub(crate) fn emit_atomic_rmw(
+        &mut self,
+        arg: &MemArg,
+        op: RmwOp,
+        size: OperandSize,
+        extend: Option<ExtendKind>,
+    ) -> Result<()> {
+        let operand = self.context.pop_to_reg(self.masm, None).unwrap();
+        if let Some(addr) = self.emit_compute_heap_address(arg, size)? {
```

Comment:
Just a note that this will require a rebase once https://github.com/bytecodealliance/wasmtime/pull/9987 lands and use `emit_compute_heap_address_align_checked`